chronos/asyncmacro2.nim

```diff
@@ -323,17 +228,38 @@ proc asyncSingleProc(prc: NimNode): NimNode {.compileTime.} =
       closureIterator.addPragma(newIdentNode("gcsafe"))
     outerProcBody.add(closureIterator)
 
-    # -> createCb(retFuture)
-    # NOTE: The "_continue" suffix is checked for in asyncfutures.nim to produce
-    # friendlier stack traces:
-    var cbName = genSym(nskVar, prcName & "_continue")
-    var procCb = getAst createCb(retFutureSym, iteratorNameSym,
-                         newStrLitNode(prcName),
-                         cbName,
-                         createFutureVarCompletions(futureVarIdents, nil))
-    outerProcBody.add procCb
-
-    # -> return retFuture
+    # -> var resultFuture = newFuture[T]()
+    # declared at the end to be sure that the closure
+    # doesn't reference it, avoid cyclic ref (#203)
+    var retFutureSym = ident "resultFuture"
+    var subRetType =
+      if returnType.kind == nnkEmpty:
+        newIdentNode("void")
+      else:
+        baseType
+    # Do not change this code to `quote do` version because `instantiationInfo`
+    # will be broken for `newFuture()` call.
+    outerProcBody.add(
```

Comment:
Is there any reason why you can't use `macros.quote` here?

Reply:
No idea, I just moved that from above